src/components/bottom-bar.tsx

```diff
@@ -17,26 +18,50 @@ interface IState {}
 export class BottomBarComponent extends BaseComponent<IProps, IState> {
 
   public render() {
-    const {ui} = this.stores;
-    const primaryMagType = ui.primaryMagnet;
-    const secondaryMagType = ui.secondaryMagnet;
-    const secondaryRowClass = secondaryMagType !== "none" ? "row" : "row hidden";
+    const {simulation} = this.stores;
+    const primaryMag = simulation.getMagnetAtIndex(0);
+    const primaryMagType = primaryMag !== null ? primaryMag.type : "none";
```

Comment:
A number of these null checks are redundant, it feels like the logic could be simplified by either including some branching logic or using `null` instead of `"none"`

Reply:
Agreed.  I felt that way when typing it out.  I didn't have a strong reason to initially use `none` (other than being influenced by C# style enums).  Probably making them nullable instead of including `none` is a better choice.

Reply:
Probably irrelevant given the previous conversation, but worth noting that `primaryMag !== null` won't catch the `undefined` case.

Reply:
Fixed.